### PR TITLE
feat(pipeline): enable key rotation

### LINKED ISSUE #23595: pipelines: not possible to enable key rotation
### Describe the bug

When creating a _modern_ pipeline (`pipelines.CodePipeline`), it should be possible to enable key rotation of the internal `cp.Pipeline`).

### Expected Behavior

There should be a property `enableKeyRotation` on a `pipelines.CodePipeline` construct.

### Current Behavior

It is not currently possible to enable key rotation on a `pipelines.CodePipeline` construct.

### Reproduction Steps

Create a new `pipelines.CodePipeline` and observe that the created KMS key does not have key rotation enabled.

### Possible Solution

Wire the `enableKeyRotation` property from `cp.Pipeline` through to `pipelines.CodePipeline`.

### Additional Information/Context

_No response_

### CDK CLI Version

2.59.0

### Framework Version

2.59.0

### Node.js Version

16.18.1

### OS

Linux

### Language

Typescript

### Language Version

4.9.3

### Other information

_No response_

### PR DESCRIPTION
**Pull request by tkglaser**
----

closes #23595 

### All Submissions:

* [y] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Construct Runtime Dependencies:

* [n] This PR adds new construct runtime dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-construct-runtime-dependencies)

### New Features

* [y] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [y] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by tkglaser** [comment]
I have tried running the integration test but getting an unspecified internal failure from CF. Not sure where to go with that, could a maintainer please try to repro?

image

File: packages/@aws-cdk/pipelines/test/codepipeline/codepipeline.test.ts [hunk 1]
```diff
@@ -192,6 +192,36 @@ test('CodeBuild action role has the right AssumeRolePolicyDocument', () => {
...
+    repositoryName: 'MyRepo',
+  });
+  const cdkInput = cdkp.CodePipelineSource.codeCommit(
+    repo,
+    'main',
+  );
+
+  new CodePipeline(pipelineStack, 'Pipeline', {
+    enableKeyRotation: true,
+    crossAccountKeys: true, // requirement of key rotation
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/pipelines/test/codepipeline/codepipeline.test.ts:207]: Given this dependency, this should be enforced in code. If someone adds rotation without setting this to true, what's the behavior? We don't want a deploy time failure or a silent failure here.

File: packages/@aws-cdk/pipelines/test/integ.newpipeline-with-cross-account-keys.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/pipelines/test/integ.newpipeline-with-cross-account-keys.ts:1]: Which dependency is this impacting?

File: packages/@aws-cdk/pipelines/test/integ.newpipeline-with-cross-account-keys.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+/// !cdk-integ PipelineStack pragma:set-context:@aws-cdk/core:newStyleStackSynthesis=true
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/pipelines/test/integ.newpipeline-with-cross-account-keys.ts:2]: This is no longer relevant when using the IntegTest construct for your tests. They always use the new style stack synthesis.

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Thank you for your contribution! I have just a few comments inline.

File: packages/@aws-cdk/pipelines/test/codepipeline/codepipeline.test.ts (hunk 1 above)
Comment (tkglaser) [reply packages/@aws-cdk/pipelines/test/codepipeline/codepipeline.test.ts:207]: This is already enforced by the underlying `Pipeline` construct. Will add a unit test to capture that.

File: packages/@aws-cdk/pipelines/test/integ.newpipeline-with-cross-account-keys.ts (hunk 2 above)
Comment (tkglaser) [reply packages/@aws-cdk/pipelines/test/integ.newpipeline-with-cross-account-keys.ts:1]: Copy and paste error, will remove.

File: packages/@aws-cdk/pipelines/test/integ.newpipeline-with-cross-account-keys.ts (hunk 3 above)
Comment (tkglaser) [reply packages/@aws-cdk/pipelines/test/integ.newpipeline-with-cross-account-keys.ts:2]: Copy and paste error, will remove.

**Review by TheRealAmazonKendra — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**